### ISSUE TITLE
Cannot access class properties on instances

### ISSUE THREAD

**Issue by geigerzaehler** [microsoft/pyright#6555]
I have the following code

```python
# pyright: strict

class Foo:
    @classmethod
    @property
    def bar(cls) -> int:
        return 0

Foo().bar  # <- type error
```

I expect this to not have any type errors. But when I run `pyright` on this file I see the following
```
/redacted/foo.py
  /redacted/foo.py:11:7 - error: Cannot access member "bar" for type "Foo"
    Failed to call method "__get__" for descriptor class "property" (reportGeneralTypeIssues)
  /redacted/foo.py:11:1 - error: Type of "bar" is unknown (reportUnknownMemberType)
2 errors, 0 warnings, 0 informations 
```

I’m using version 1.1.337 and this broke in 1.1.335.

**Comment by erictraut** [comment]
Class properties require some especially ugly hacks in a type checker — as they do in the runtime. Mypy doesn't support them at all. They were introduced in Python 3.9, deprecated in 3.11, and will be removed completely in 3.13. If your code is using them, you should look at moving away from them.

I'm half inclined to simply remove support for class properties entirely in pyright. Perhaps it would be best to emit a deprecation warning for now.

I'll see if there's a straightforward way to fix this regression in pyright.

**Comment by geigerzaehler** [comment]
Thanks for the heads-up. I wasn’t aware that wrapping descriptors with `classmethod` [was deprecated](https://docs.python.org/3.11/whatsnew/3.11.html#language-builtins). I’ll update my code so I won’t have a need for this anymore.

**Comment by erictraut** [comment]
I've fixed the regression, and I've also added a deprecation warning (which is an error in strict mode) when a class property is defined. This will be included in the next release of pyright.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.338, which I just published. It will also be included in a future release of pylance.
